tutor/src/screens/assignment-grade/ux.js

```diff
@@ -90,29 +100,62 @@ export default class AssignmentGradingUX {
     this.viewedQuestions.set(index, true);
   }
 
-  @action async saveScore({ response, points, comment }) {
+  isResponseGraded(response) {
+    return response.grader_points !== undefined;
+  }
+
+  @action async saveScore({ points, comment, response, doGoToOverview = false, doMoveNextQuestion = false }) {
     const grade = new Grade({ points, comment, response });
     await grade.save();
     //refetch scores after grade was saved
     await this.planScores.fetch();
+    await this.planScores.taskPlan.fetch();
+    await this.planScores.taskPlan.analytics.fetch();
     await this.planScores.ensureExercisesLoaded();
-    
+
+    if(doGoToOverview) 
+      this.goToOverview();
     // move to next question if any
-    if (isEmpty(this.unGraded)) {
+    else if (isEmpty(this.unGraded) && doMoveNextQuestion) {
       if (this.selectedHeadingIndex < this.headings.length - 1)
         this.selectedHeadingIndex += 1;
     }
```

Comment:
I usually prefer to always use braces, otherwise it's really easy to add a new line right before `this.goToOverview();` and break it.